components/zoho_desk/zoho_desk.app.mjs

```diff
@@ -1,11 +1,354 @@
+    getDepartments(args = {}) {
+      return this.makeRequest({
+        path: "/departments",
+        ...args,
+      });
+    },
+    createTicket(args = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: "/tickets",
+        ...args,
+      });
+    },
+    updateTicket({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "patch",
+        path: `/tickets/${ticketId}`,
+        ...args,
+      });
+    },
+    createTicketComment({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/tickets/${ticketId}/comments`,
+        ...args,
+      });
+    },
+    getTickets(args = {}) {
+      return this.makeRequest({
+        path: "/tickets",
+        ...args,
+      });
+    },
+    searchTickets(args = {}) {
+      return this.makeRequest({
+        path: "/tickets/search",
+        ...args,
+      });
+    },
+    createTicketAttachment({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/tickets/${ticketId}/attachments`,
+        ...args,
+      });
+    },
+    getTicketAttachments({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        path: `/tickets/${ticketId}/attachments`,
+        ...args,
+      });
+    },
+    getTicketComments({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        path: `/tickets/${ticketId}/comments`,
+        ...args,
+      });
+    },
+    sendReply({
+      ticketId, ...args
+    } = {}) {
+      return this.makeRequest({
+        method: "post",
+        path: `/tickets/${ticketId}/sendReply`,
+        ...args,
+      });
+    },
+    getContacts(args = {}) {
+      return this.makeRequest({
+        path: "/contacts",
+        ...args,
+      });
+    },
+    getSupportEmailAddresses(args = {}) {
+      return this.makeRequest({
+        path: "/supportEmailAddress",
+        ...args,
+      });
+    },
+    getAccounts(args = {}) {
+      return this.makeRequest({
+        path: "/accounts",
+        ...args,
+      });
+    },
+    getAgents(args = {}) {
+      return this.makeRequest({
+        path: "/agents",
+        ...args,
+      });
+    },
+    async getResourcesOptions({
+      orgId, departmentId, prevContext, resourceFn, resourceMapper,
+    }) {
+      const { from = 1 } = prevContext;
+      if (from === null) {
+        return [];
+      }
+      const { data: resources } =
+        await resourceFn({
+          headers: {
+            orgId,
+          },
+          params: {
+            departmentId,
+            from,
+            limit: constants.DEFAULT_LIMIT,
+          },
+        });
+      const currentLen = resources.length;
+      const options = resources.map(resourceMapper);
+      return {
+        options,
+        context: {
+          index: currentLen
+            ? currentLen + from
+            : null,
+        },
+      };
```

Comment:
I think `index` here is supposed to be `from`. I don't see `from` being set anywhere else.
```suggestion
      return {
        options,
        context: {
          from: currentLen
            ? currentLen + from
            : null,
        },
      };
```